packages/components/src/sites-table/site-status.tsx

```diff
@@ -0,0 +1,35 @@
+import { useI18n } from '@wordpress/react-i18n';
+
+export type SiteStatus = 'coming-soon' | 'private' | 'public';
+
+export interface SiteObjectWithStatus {
+	is_coming_soon?: boolean;
+	is_private?: boolean;
+	launch_status?: string;
+}
+
+export const getSiteStatus = ( site: SiteObjectWithStatus ): SiteStatus => {
+	if ( site.is_coming_soon || ( site.is_private && site.launch_status === 'unlaunched' ) ) {
+		return 'coming-soon';
+	}
+
+	if ( site.is_private ) {
+		return 'private';
+	}
+
+	return 'public';
+};
+
+export const useTranslatedSiteStatus = ( site: SiteObjectWithStatus ) => {
```

Comment:
I've decided to split `useSiteStatus` for reusability and redundancy reduction.

`useTranslatedSiteStatus` now does not return the site status (it's not necessary in all cases). We can use `getSiteStatus` in conjunction with it if we want to hide the label in some instances (i.e. site is public in tile mode).

Reply:
I like the split: the vanilla JS function to do the pure JS stuff, and the hook to do the fancy React stuff.

Another option? `Label` is less of a mouthful and makes it clear how the string can be used and that it will be translated (because all labels are translated).
```suggestion
export const useSiteStatusLabel = ( site: SiteObjectWithStatus ) => {
```